### PR TITLE
feat: Change to Microsoft.Extensions.Configuration.Ini resolution .conf

### PR DESCRIPTION
**Pull request by Tanyuu**
Since the library can't handle newlines and comments, the ParseStringInit method processes the string first and then feeds it into ParseBuffer. in addition, since the library doesn't support net452, it retains the old parse method.

Fix: https://github.com/casbin/Casbin.NET/issues/264

### CODE REVIEW

**Comment by hsluoyz** [comment]
@sagilio

File: Casbin/Config/DefaultConfig.cs [hunk 1]
```diff
@@ -159,26 +164,51 @@ private bool AddConfig(string section, string option, string value)
...
             }
+            if (processedValue != string.Empty)
+            {
+                textWriter.WriteLine(processedValue);
+            }
+            return new MemoryStream(Encoding.UTF8.GetBytes(textWriter.ToString()));
         }
 
-        private void ParseBuffer(TextReader reader)
+        private void ParseBuffer(Stream reader)
```
Comment (sagilio) [inline Casbin/Config/DefaultConfig.cs:196]:
```suggestion
        private void AddStream(Stream steam)
```

File: Casbin/Config/DefaultConfig.cs [hunk 2]
```diff
@@ -159,26 +164,51 @@ private bool AddConfig(string section, string option, string value)
 
         private void Parse(string configFilePath)
         {
-            using (var sr = new StreamReader(configFilePath))
+            ParseBuffer(ParseStringInit(new StreamReader(configFilePath)));
+        }
+
+        private MemoryStream ParseStringInit(TextReader textReader)
```
Comment (sagilio) [inline Casbin/Config/DefaultConfig.cs:170]:
```suggestion
        private Stream RemoveComment(Stream stream)
```
Plz try to design by this method signature, and be called from the `AddStream` method

File: Casbin/Config/DefaultConfig.cs [hunk 3]
```diff
@@ -159,26 +164,51 @@ private bool AddConfig(string section, string option, string value)
 
         private void Parse(string configFilePath)
         {
-            using (var sr = new StreamReader(configFilePath))
+            ParseBuffer(ParseStringInit(new StreamReader(configFilePath)));
```
Comment (sagilio) [inline Casbin/Config/DefaultConfig.cs:167]:
```suggestion
            using FileStream fileStream = File.OpenRead(filePath);
            AddStream(fileStream);
```

File: Casbin/Config/DefaultConfig.cs [hunk 4]
```diff
@@ -48,7 +53,7 @@ public static IConfig CreateFromFile(string configFilePath)
         public static IConfig CreateFromText(string text)
         {
             var config = new DefaultConfig();
-            config.ParseBuffer(new StringReader(text));
+            config.ParseBuffer(config.ParseStringInit(new StringReader(text)));
```
Comment (sagilio) [inline Casbin/Config/DefaultConfig.cs:56]:
```suggestion
            using var memoryStream = new MemoryStream(text);
            config.AddStream(memoryStream);
```

**Review by sagilio — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@sagilio plz review

File: Casbin/Casbin.csproj [hunk 5]
```diff
@@ -37,77 +37,86 @@
...
+    <PackageReference Include="System.Memory" Version="4.5.5" />
+    <PackageReference Include="JetBrains.Annotations" Version="2023.2.0" PrivateAssets="All" />
+    <PackageReference Include="Microsoft.SourceLink.Github" Version="1.1.1" PrivateAssets="All" />
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFramework)' == 'net8.0'">
-    <PackageReference Include="DynamicExpresso.Core" Version="2.16.1"/>
-    <PackageReference Include="Microsoft.Extensions.Logging" Version="8.0.0-preview.5.23280.8"/>
+    <PackageReference Include="DynamicExpresso.Core" Version="2.16.1" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Ini" Version="7.0.0" />
```
Comment (sagilio) [inline Casbin/Casbin.csproj:54]: The version is wrong, plz change the version according to the target version.

File: Casbin/Config/DefaultConfig.cs [hunk 6]
```diff
@@ -159,26 +167,53 @@ private bool AddConfig(string section, string option, string value)
...
+            using FileStream fileStream = File.OpenRead(configFilePath);
+            AddStream(RemoveComment(fileStream));
+        }
+
+        private Stream RemoveComment(Stream stream)
+        {
+            TextWriter textWriter = new StringWriter();
+            string line;
+            string processedValue = string.Empty;
+            StreamReader streamreader = new StreamReader(stream);
```
Comment (sagilio) [inline Casbin/Config/DefaultConfig.cs:179]:
```suggestion
            using var streamReader = new StreamReader(stream);
```

File: Casbin/Config/DefaultConfig.cs [hunk 7]
```diff
@@ -159,26 +167,53 @@ private bool AddConfig(string section, string option, string value)
...
+        private void AddStream(Stream steam)
         {
+#if NET452
             string section = string.Empty;
             int lineNum = 0;
             string line;
             bool inSuccessiveLine = false;
             string option = string.Empty;
             string processedValue = string.Empty;
+            TextReader textReader = new StreamReader(steam);
```
Comment (sagilio) [inline Casbin/Config/DefaultConfig.cs:210]:
```suggestion
            using var streamReader = new StreamReader(stream);
```

**Review by sagilio — CHANGES_REQUESTED** [review]

**Review by sagilio — APPROVED** [review]
LGTM

**Pull request merged**